fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java

```diff
@@ -302,14 +258,10 @@ private void refreshTable(BaseTable updatedTable, long baseSnapshotId,
         IcebergTableName updatedIcebergTableName = new IcebergTableName(dbName, tableName, updatedSnapshotId);
         long latestRefreshTime = tableLatestRefreshTime.computeIfAbsent(new IcebergTableName(dbName, tableName), ignore -> -1L);
 
-        List<String> updatedPartitionNames = updatedTable.spec().isPartitioned() ?
-                listPartitionNamesWithSnapshotId(updatedTable, dbName, tableName, updatedSnapshotId, executorService) :
-                new ArrayList<>();
-
+        partitionCache.invalidate(baseIcebergTableName);
+        partitionCache.getUnchecked(updatedIcebergTableName);
```

Comment:
Map result = delegate.loadPartitions(key.dbName, key.tableName, key.snapshotId,executorService);
partitionCache.put(updatedIcebergTableName, result)

Reply:
partitionCache is a async loading cache. When we fetch it, it will load data into cache.